### ISSUE TITLE
support TechDocs edit button for Bitbucket Cloud

### ISSUE THREAD

**Issue by pjungermann** [backstage/backstage#12892]
## Feature Suggestion

Add support for the edit button at TechDocs pages.

## Possible Implementation

See PR #12857 which was adding it for Bitbucket Server.

**Comment by pjungermann** [comment]
@awanlin maybe something interesting for Azure DevOps, too?

**Comment by awanlin** [comment]
Yeah, would be nice to have, unfortunately the way we build TechDocs has be come a little bespoke so I'm not sure if this is something I could contribute

**Comment by pjungermann** [comment]
I looked into this. The way how support was done for Bitbucket Server is not optimal as it drops edit support for other places like at the refs to the catalog files.

The root cause / issue lies at mkdocs as it does not support query parameters well. Unfortunately, there is no intention to change this. However, it is possible to have a plugin to handle these cases. This could be integrated (directly or via another dependency/plugin) at the techdocs-core plugin for it. 🤔 

Until then, it may make sense to just cut off the query parameters as standard logic which could apply to all implementations which need query parameters. This would mean that the link is not a direct link into the online editor / active edit mode, but at least the file itself at which you can enter the edit mode yourself easily.

Better than nothing.

This would bring back the edit option for catalog files for Bitbucket Server, too.
